google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTracerTest.java

```diff
@@ -702,6 +709,29 @@ public void testPermanentFailure() {
     verifyAttributes(opLatency, expected);
   }
 
+  @Test
+  public void testRemainingDeadline() {
```

Comment:
I think we can calculate what the remaining deadline would be after retries. for example in the retry settings we can set initialRpcTimeout=300, rpcTimeoutMutliplier=1, maxRpcTimeout=300, totalTimeout=500, initialBackoffTime=10, backoffTimeMultipler=1, maxBackoffTime=10, and let the server retry the request 1 time, the remaining deadline should = totalTimeout - initialRpcTimeout - initialBackofftime = 190.